src/ImpactX.cpp

```diff
@@ -64,6 +68,23 @@ namespace impactx
         //   before we start the evolve loop, we are in "step 0" (initial state)
         int global_step = 0;
 
+        // check typos in inputs after step 1
+        bool early_params_checked = false;
+
+        // count particles - if no particles are found in our particle container, then a lot of
```

Comment:
Please remove this - this should now be part of `validate()` already, isn't it? Otherwise please move there.